Canned Autoresponse

greezybacon opened the pull request:
I merged the previous PR by mistake - I meant to add the following comments;
1. (Automatically respond with this canned attachment) - > attachment => response?
2. Make sure we restrict deletion of in-use canned response in knowledge base (check filters)

greezybacon commented:
I think I'm missing something. Are these additional feature requests? I don't see how they merge into previous PR.

protich commented:
1. is a typo on filter page - it should say _canned response_ not _canned attachment_.
2. is sort of needed so staff can't unknowingly delete a canned response that is used by a filter.

It ties in to the previous PR - I just forgot to comment and ended up merging.

greezybacon commented:
Cool. Too late for tonight. I'll try and have it for tomorrow night.

greezybacon committed:
Fix email filter -> canned response reference

greezybacon committed:
Prohibit disabling and deleting canned responses
when they are in use by email filters

greezybacon committed:
Indicate which filters are using a canned reply

protich commented on include/staff/cannedreplies.inc.php:
```diff
@@ -81,9 +83,12 @@
                 ?>
             <tr id="<?php echo $row['canned_id']; ?>">
                 <td width=7px>
+                <?php if (!$row['in_use']) { ?>
```
Does this means it can't be disabled - when in-use by a filter?

I'll make a case that it's OK to allow admin to DISABLE it as away to turn off a canned-auto-response used by multiple filters for example.

The proper place to protect against deletion is `$canned->delete()` function. Do a join on load and simply do `if($this->isInUse()) return false;`

greezybacon committed:
Allow disabling canned responses that are used by email filters

But don't send out canned responses for new tickets that matched a filter
with an attached canned response if that response is not enabled. Also,
explicitly block deletes to filters that are in use.

protich commented:
Having the actual count of filters might be helpful - we can use it on canned response page as part of the warning.

greezybacon committed:
Reuse list of filters which specify a canned response

greezybacon committed:
Keep number of associated filters in canned reply info

greezybacon committed:
Don't show in-use warning for new canned replies

greezybacon committed:
No need for distinct, unique field
